src/charm.py

```diff
@@ -584,6 +585,16 @@ def _update_cert(self):
         self.container.exec(["update-ca-certificates", "--fresh"]).wait()
         subprocess.run(["update-ca-certificates", "--fresh"])
 
+    def _check_disk_space(self):
+        try:
+            database_storage = self.model.storages.get('database', [])
+            if database_storage and shutil.disk_usage(database_storage[0].location).free < 1e9: # type: ignore
+                self._stored.status["disk_space"] = to_tuple(BlockedStatus("Less than 1 Gi remaining in storage"))
+
+        # If this check is done before storage is attached, we don't want the charm to go error state
+        except FileNotFoundError:
+            self.unit.status = BlockedStatus("Storage not available")
```

Comment:
Should we use the same error namespace as before? And should 

```suggestion
            self._stored.status["disk_space"] = MaintenanceStatus("Storage not available")
```